mealpy/mealpy.py

```diff
@@ -204,19 +207,47 @@ def cli_list():  # pragma: no cover
 
 @cli_list.command('cities', short_help='List available cities.')
 def cli_list_cities():  # pragma: no cover
-    cities = [i['name'] for i in MealPal.get_cities()]
-    print('\n'.join(cities))
+    print('\n'.join(list_cities()))
+
+
+def list_cities():
+    cities_file = CACHE_HOME / 'cities.json'
+    cities = [city['name'] for city in cache_list(cities_file, MealPal.get_cities)]
+
+    return cities
 
 
 @cli_list.command('restaurants', short_help='List available restaurants.')
 @click.argument('city')
 def cli_list_restaurants(city):  # pragma: no cover
-    restaurants = [i['restaurant']['name'] for i in MealPal.get_schedules(city)]
+    restaurants = [i['restaurant']['name'] for i in list_menu(city)]
     print('\n'.join(restaurants))
 
 
 @cli_list.command('meals', short_help='List meal choices.')
 @click.argument('city')
 def cli_list_meals(city):  # pragma: no cover
-    restaurants = [i['meal']['name'] for i in MealPal.get_schedules(city)]
+    restaurants = [i['meal']['name'] for i in list_menu(city)]
     print('\n'.join(restaurants))
+
+
+def list_menu(city):  # pragma: no cover
+    menu_file = CACHE_HOME / 'menu.json'
+    callback = partial(MealPal.get_schedules, city)
+    return cache_list(menu_file, callback)
+
+
+def cache_list(cache_file, callback):
+    items = []
+
+    if cache_file.exists():
+        data = json.load(cache_file.open())
+        cache_expire_date = pendulum.parse(data['run_date']).add(hours=1)
```

Comment:
I'm not sure what the behavior would be if someone requests for the menu one minute before the next day's kitchen opens. Assuming that some result is returned and is cached (not sure if anything is even returned when kitchen is closed), when the kitchen eventually opens the cache still contains the old results if expiry is one hour.

Reply:
Another good catch. It would cache the "kitchen closed" response for 1 hour. Later today, I'll observe what the response and only write to cache for successful responses.

Reply:
I think there might still be an issue across cities since kitchen close times are different.

Reply:
Yeah, it'll need to be cached per-city. So what we have, but a list of them instead. i.e.
```json
[
  { "run_date": "2019-01-01",
    "city": "SF",
    "results": {...}
  },
  { "run_date": "2018-01-01",
    "city": "LA",
    "results: {...}
  }
]
```

Reply:
> I'm not sure what the behavior would be if someone requests for the menu one minute before the next day's kitchen open

Just tested this, there is data set up for the next business day. i.e. I ran this at 19:00 on Friday, the menu choices for Monday are returned. So we don't need to worry about stale results (asides from the missing general error handling, only cache when successful).

Reply:
But it looks like there will always be stale results if you happen to cache the menu just before the menu for the next business day is being updated.

Reply:
Sure, it'll be stale for 1 hour. How do you propose we handle this? We don't actually know when the menu for the next day goes live, we've just been assuming it's 17:00.

This is a simple TTL cache (with all the pros and cons). You're proposing an additional cache invalidation criteria but it's hard to pin down what that is. Maybe invalidate at 17:00 local time?

Reply:
Then IMO the cache should only be used as a failover mechanism if there is a network connection error and some argument is used when running the script so the user knows that the result might be outdated. Avoiding unnecessary requests because of rogue scripts isn't really a concern of the mealpy script since the idea is that we just provide an interface to the data, and the user should cache their own results if they think they are at risk of this problem. There is a tradeoff between minimizing network calls and availability of latest results, and by implementing this cache on our end we are forcing the user to give up the availability of latest results without any option of choosing otherwise.

Reply:
That sounds fair. This is well within the realm of premature optimizations. I'll discard.